custom_components/fraimic/image_convert.py

```
@@ -209,6 +229,139 @@ def _preprocess(
+        image = image.filter(ImageFilter.SMOOTH)
+        image = image.filter(ImageFilter.SHARPEN)
+
+    if pad_color is not None and image.size != size:
+        framed = Image.new("RGB", size, pad_color)
+        framed.paste(image, ((width - image.width) // 2, (height - image.height) // 2))
+        return framed
+    return image
+
+
+@cache
+def _official_palette_lut():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_official_palette_lut`

(ANN202)